diff -ruN lvgl/demos/ebike/lv_demo_ebike_private.h lvgl-orig/demos/ebike/lv_demo_ebike_private.h
--- lvgl/demos/ebike/lv_demo_ebike_private.h	2024-10-20 01:47:28.859970603 +0000
+++ lvgl-orig/demos/ebike/lv_demo_ebike_private.h	2024-10-20 01:48:21.510006690 +0000
@@ -13,7 +13,7 @@
 /*********************
  *      INCLUDES
  *********************/
-#include "../../../lvgl/lvgl.h"
+#include "../../lvgl.h"
 
 /*********************
  *      DEFINES
